policy-controller/k8s/status/src/index.rs

```diff
@@ -255,11 +297,16 @@ impl kubert::index::IndexNamespacedResource<k8s::policy::HttpRoute> for Index {
         let name = resource.name_unchecked();
         let id = ResourceId::new(namespace, name);
 
-        // Create the route parents and insert it into the index. If the
-        // HTTPRoute is already in the index and it hasn't changed, skip
-        // creating a patch.
-        let parents = http_route::make_parents(resource);
-        if !self.update_http_route(id.clone(), parents.clone()) {
+        // Create the route parents
+        let parents = http_route::make_parents(resource.clone());
+
+        // Create the route backends
+        let backends = http_route::make_backends(resource);
+
+        // Construct references and insert into index; if HTTPRoute is already
+        // in the index and it hasn't changed, skip patch.
+        let references = References { parents, backends };
+        if !self.update_http_route(id.clone(), references.clone()) {
```

Comment:
minor efficiency nit: perhaps `update_http_route` could take `References` by reference (an `&References`) and only clone it if we're actually inserting it into the hashmap?